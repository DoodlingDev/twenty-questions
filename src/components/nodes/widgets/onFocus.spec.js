/* eslint require-jsdoc: "off" */
import React from "react";
import OnFocus from "./onFocus";
import { /* shallow,*/ mount } from "enzyme";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const defaultTestProps = {
  name: "test_onfocus",
  path: "test_path",
  properties: [
    {
      type: "string",
      name: "test_string",
    },
  ],
  valueManager: {
    values: {},
    validate: {},
  },
};

function setup(renderFn, testProps = defaultTestProps) {
  return renderFn(<OnFocus {...testProps} />);
}

describe("rendering", () => {
  it("renders correctly", () => {
    const render = renderer.create(<OnFocus {...defaultTestProps}/>).toJSON();
    expect(render).toMatchSnapshot();
  });
});

describe("mounted component", () => {
  it("should render the label view by default", () => {
    const wrapper = setup(mount);
    expect(wrapper.find("input").length < 1).toBe(true);
  });

  it("should render the label view if unfocused", () => {
    const wrapper = setup(mount);
    wrapper.state.focused = false;
    wrapper.update();
    expect(wrapper.find("input").length < 1).toBe(true);
  });

  it("should render the inputs view if focused", () => {
    const wrapper = setup(mount);
    wrapper.setState({ focused: true });
    wrapper.update();
    expect(wrapper.find("input").length < 1).toBe(false);
  });
});

describe("with existing data", () => {
  const existingDataProps = {
    name: "test_onfocus",
    path: "test_path",
    properties: [
      {
        type: "string",
        name: "test_string",
      },
    ],
    valueManager: {
      values: {
        "test_path.test_string": "existing data",
      },
      validate: {},
    },
  };

  it("should render the input view", () => {
    const wrapper = setup(mount, existingDataProps);
    expect(wrapper.find("input").length >= 1).toBe(true);
  });
});
