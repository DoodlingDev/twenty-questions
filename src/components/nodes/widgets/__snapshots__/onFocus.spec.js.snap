// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="q20_focusField__testPath--awaitingFocus"
  onFocus={[Function]}
  tabIndex="0"
>
  test_onfocus
</div>
`;
